js/visualization_entity_charts_steps.js

```diff
@@ -355,14 +355,11 @@ this.recline.View.nvd3 = this.recline.View.nvd3 || {};
         backend: backend,
         url: url
       };
-      
+
       state.set('source', source);
-      $.get(url).done(function(data){
-        data = data.replace(/(?:\r|\n)/g, '\r\n');
-        data = CSV.parse(data);
-        state.set('model', new recline.Model.Dataset({records: data}));
-        cb(state);
-      });
+      var model = new recline.Model.Dataset(source);
+      state.set('model', model)
+      model.fetch().done(cb.bind(this, state));
```

Comment:
Do we need to catch a broken promise here?

+        model.fetch().done(graph.render.bind(graph)).fail(function (err) { console.log('Failed to fetch', err); });
 
Or alert? Not sure if there is a convention in place.